src/index.ts

```diff
@@ -1,30 +1,462 @@
 import {
-  JupyterFrontEnd,
-  JupyterFrontEndPlugin
+  ILabShell,
+  JupyterFrontEndPlugin,
+  IRouter,
+  JupyterFrontEnd
 } from '@jupyterlab/application';
+import { showDialog, Dialog } from '@jupyterlab/apputils';
+import {
+  IDefaultFileBrowser,
+  IFileBrowserFactory,
+  FileBrowser,
+  Uploader
+} from '@jupyterlab/filebrowser';
+
+import {
+  createToolbarFactory,
+  IToolbarWidgetRegistry,
+  setToolbar
+} from '@jupyterlab/apputils';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+import { ITranslator } from '@jupyterlab/translation';
+
+import { CommandRegistry } from '@lumino/commands';
+import { Widget } from '@lumino/widgets';
+import { Drive } from './s3contents';
+
+import { DriveIcon } from './icons';
+import { FilenameSearcher, IScore } from '@jupyterlab/ui-components';
+import { Token } from '@lumino/coreutils';
+import { S3ClientConfig } from '@aws-sdk/client-s3';
 
-import { defaultFileBrowser, toolbarFileBrowser } from './filebrowser';
+/**
+ * The command IDs used by the filebrowser plugin.
+ */
+namespace CommandIDs {
+  export const openPath = 'filebrowser:open-path';
+  export const openChangeDrive = 'drives:open-change-drive-dialog';
+  export const copyToAnotherBucket = 'drives:copy-to-another-bucket';
+}
+
+const FILE_BROWSER_FACTORY = 'DriveBrowser';
+const FILE_BROWSER_PLUGIN_ID = 'jupyter-drives-browser:file-browser-toolbar';
 
 /**
- * Initialization data for the jupyter-drives-browser extension.
+ * The class name added to the  drive filebrowser filterbox node.
  */
-const defaultPlugin: JupyterFrontEndPlugin<void> = {
-  id: 'jupyter-drives-browser:plugin',
-  description:
-    'A JupyterLab extension which enables client-side drives access.',
+const FILTERBOX_CLASS = 'jp-DriveBrowser-filterBox';
+
+/**
+ * The class name added to file dialogs.
+ */
+const FILE_DIALOG_CLASS = 'jp-FileDialog';
+
+/**
+ * The class name added for the new drive label in the switch drive dialog.
+ */
+const SWITCH_DRIVE_TITLE_CLASS = 'jp-new-drive-title';
+
+/**
+ * An interface to describe S3 authentication provided by a plugin.
+ */
+interface IS3Auth {
+  bucket: string;
+  config: S3ClientConfig;
+}
+
+/**
+ * A token for a plugin that provides S3 authentication.
+ */
+const IS3Auth = new Token<IS3Auth>('jupyter-drives-browser:auth-file-browser');
+
+/**
+ * The auth/credentials provider for the file browser.
+ */
+const authFileBrowser: JupyterFrontEndPlugin<IS3Auth> = {
+  id: 'jupyter-drives-browser:auth-file-browser',
+  description: 'The default file browser auth/credentials provider',
+  provides: IS3Auth,
+  activate: async (): Promise<IS3Auth> => {
+    return {
+      bucket: 'jupyter-drives-test-bucket-1',
+      config: {
+        forcePathStyle: true,
+        endpoint: 'https://example.com/s3',
+        region: 'eu-west-1',
+        credentials: {
+          accessKeyId: 'abcdefghijklmnopqrstuvwxyz',
+          secretAccessKey: 'SECRET123456789abcdefghijklmnopqrstuvwxyz'
```

Comment:
Could you support also `sessionToken` like in [this example](https://supabase.com/docs/guides/storage/s3/authentication)? 

```js
import { S3Client } from '@aws-sdk/client-s3'

const {
  data: { session },
} = await supabase.auth.getSession()

const client = new S3Client({
  forcePathStyle: true,
  region: 'project_region',
  endpoint: 'https://project_ref.supabase.co/storage/v1/s3',
  credentials: {
    accessKeyId: 'project_ref',
    accessSecretKey: 'anonKey',
    sessionToken: session.access_token,
  },
})
```

Reply:
Yes, `sessionToken` is supported because it is an optional field of `S3ClientConfig`